lib/presentation/widget/biometric_authentication/biometric_authentication_setting_viewmodel.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:linshare_flutter_app/presentation/util/router/app_navigation.dart';
+import 'package:linshare_flutter_app/presentation/widget/base/base_viewmodel.dart';
+import 'package:redux/redux.dart';
+import 'package:linshare_flutter_app/presentation/util/extensions/list_biometric_kind_extension.dart';
+
+class BiometricAuthenticationSettingViewModel extends BaseViewModel {
+  final AppNavigation _appNavigation;
+  final AuthenticationBiometricInteractor _authenticationBiometricInteractor;
+  final EnableBiometricInteractor _enableBiometricInteractor;
+  final GetAvailableBiometricInteractor _getAvailableBiometricInteractor;
+  final GetBiometricSettingInteractor _getBiometricSettingInteractor;
+  final DisableBiometricInteractor _disableBiometricInteractor;
+
+  BiometricAuthenticationSettingViewModel(
+    Store<AppState> store,
+    this._appNavigation,
+    this._authenticationBiometricInteractor,
+    this._enableBiometricInteractor,
+    this._getAvailableBiometricInteractor,
+    this._getBiometricSettingInteractor,
+    this._disableBiometricInteractor
+  ) : super(store);
+
+  void getBiometricSetting() {
+    store.dispatch((Store<AppState> store) async {
+
+      await Future.wait([
+        _getBiometricSettingInteractor.execute(),
+        _getAvailableBiometricInteractor.execute()
+      ]).then((response) async {
+
+        final biometricState = response[0]
+            .map((result) => result is GetBiometricSettingViewState ? result.biometricState : BiometricState.disabled)
+            .getOrElse(() => BiometricState.disabled);
+
+        response[1].fold((failure) {
+          store.dispatch(SetBiometricAuthenticationAction(biometricState, []));
+        }, (success) {
+          store.dispatch(SetBiometricAuthenticationAction(
+              biometricState,
+              success is GetAvailableBiometricViewState ? success.biometricKinds : []));
+        });
+      });
+    });
+  }
+
+  void backToAccountDetail() {
+    _appNavigation.popBack();
+    if (store.state.biometricAuthenticationSettingState.authenticationBiometricState == AuthenticationBiometricState.unEnrolled) {
+      store.dispatch(SetAuthenticationBiometricStateAction(AuthenticationBiometricState.unAuthenticated));
+    }
+  }
+
+  void checkBiometricState(BuildContext context) {
+    final authenticationBiometricState = store.state.biometricAuthenticationSettingState.authenticationBiometricState;
+    if (authenticationBiometricState == AuthenticationBiometricState.unEnrolled) {
+      _authenticationBiometric(context, BiometricState.disabled);
+    }
+  }
+
+  void toggleBiometricState(BuildContext context) {
+    final biometricState = store.state.biometricAuthenticationSettingState.biometricState;
+    _authenticationBiometric(context, biometricState);
+  }
+
+  void _authenticationBiometric(BuildContext context, BiometricState biometricState) {
+    store.dispatch((Store<AppState> store) async {
```

Comment:
Same, you already dispatch actions at the end of this function

Reply:
I do not think so. I just want it to run in redux's thread.